### ISSUE TITLE
Failed to handle network changes (e.g., VPN disconnect)

### ISSUE THREAD

**Issue by hosiet** [bigeagle/danmaQ#24]
No danmaku will be displayed after network status changes.

```bash
% /usr/bin/danmaQ
QLayout: Attempting to add QLayout "" to DMApp "", which already has a layout
QLayout: Attempting to add QLayout "" to DMApp "", which already has a layout
QLayout: Attempting to add QLayout "" to DMApp "", which already has a layout
QLayout: Attempting to add QLayout "" to DMApp "", which already has a layout
"在路径 org.freedesktop.DBus.Properties 的对象上没有 /org/freedesktop/NetworkManager/ActiveConnection/5 接口"
"在路径 org.freedesktop.DBus.Properties 的对象上没有 /org/freedesktop/NetworkManager/ActiveConnection/4 接口"
"在路径 org.freedesktop.DBus.Properties 的对象上没有 /org/freedesktop/NetworkManager/ActiveConnection/5 接口"
"在路径 org.freedesktop.DBus.Properties 的对象上没有 /org/freedesktop/NetworkManager/ActiveConnection/4 接口"
"在路径 org.freedesktop.DBus.Properties 的对象上没有 /org/freedesktop/NetworkManager/ActiveConnection/5 接口"
"在路径 org.freedesktop.DBus.Properties 的对象上没有 /org/freedesktop/NetworkManager/ActiveConnection/4 接口"
"在路径 org.freedesktop.DBus.Properties 的对象上没有 /org/freedesktop/NetworkManager/ActiveConnection/5 接口"
"在路径 org.freedesktop.DBus.Properties 的对象上没有 /org/freedesktop/NetworkManager/ActiveConnection/4 接口"
"在路径 org.freedesktop.DBus.Properties 的对象上没有 /org/freedesktop/NetworkManager/ActiveConnection/5 接口"
"在路径 org.freedesktop.DBus.Properties 的对象上没有 /org/freedesktop/NetworkManager/ActiveConnection/4 接口"
"在路径 org.freedesktop.DBus.Properties 的对象上没有 /org/freedesktop/NetworkManager/ActiveConnection/5 接口"
"在路径 org.freedesktop.DBus.Properties 的对象上没有 /org/freedesktop/NetworkManager/ActiveConnection/4 接口"
"在路径 org.freedesktop.DBus.Properties 的对象上没有 /org/freedesktop/NetworkManager/ActiveConnection/6 接口"
"在路径 org.freedesktop.DBus.Properties 的对象上没有 /org/freedesktop/NetworkManager/ActiveConnection/6 接口"
"在路径 org.freedesktop.DBus.Properties 的对象上没有 /org/freedesktop/NetworkManager/ActiveConnection/7 接口"
"在路径 org.freedesktop.DBus.Properties 的对象上没有 /org/freedesktop/NetworkManager/ActiveConnection/6 接口"
"在路径 org.freedesktop.DBus.Properties 的对象上没有 /org/freedesktop/NetworkManager/ActiveConnection/7 接口"
"在路径 org.freedesktop.DBus.Properties 的对象上没有 /org/freedesktop/NetworkManager/ActiveConnection/6 接口"
"在路径 org.freedesktop.DBus.Properties 的对象上没有 /org/freedesktop/NetworkManager/ActiveConnection/7 接口"
"在路径 org.freedesktop.DBus.Properties 的对象上没有 /org/freedesktop/NetworkManager/ActiveConnection/6 接口"
"在路径 org.freedesktop.DBus.Properties 的对象上没有 /org/freedesktop/NetworkManager/ActiveConnection/7 接口"
"在路径 org.freedesktop.DBus.Properties 的对象上没有 /org/freedesktop/NetworkManager/ActiveConnection/6 接口"
```

**Comment by jiegec** [comment]
我在MacBookAir本地试的时候会出现无限Network access is disabled错误.
但这个错误可以用这个patch解决:
```diff
diff --git a/src/subscriber.cpp b/src/subscriber.cpp
index 96b7852..80d9b29 100644
--- a/src/subscriber.cpp
+++ b/src/subscriber.cpp
@@ -58,6 +58,7 @@ void Subscriber::run()
 	timeout.setSingleShot(true);
 	
 	while(1) {
+		http->setNetworkAccessible(QNetworkAccessManager::Accessible);
 		timeout.start(10000);
 		QNetworkReply *reply = http->get(request);
 		// If timeout signaled, let http request abort

```

**Comment by Harry-Chen** [comment]
Still investigating, will not fix in v0.2.3.
